### DISCUSSION TITLE
Classes in an object aren't detected

### DISCUSSION THREAD

**Discussion by 0x41-li** [tailwindlabs/tailwindcss#13603] [Help] [answered]
Classes on a className property inside an object aren't generated.

I knew that tailwind wasn't picking these up with whatever extraction algorithm they use, I confirmed that by replacing the colon with an equal sign, and then vscode recognized them as tailwind classes, but obviously that's not the solution as it is a syntax error on javascript.

# What I'm using:
React, typescript, vite, and tailwind. 

All with the latest versions.

# Example Code:
```js
  return React.createElement(
    TagName,
    {
      className:
        "relative mx-auto inline-block h-[38px] font-bold leading-7 tracking-[2.29px] md:leading-[26px] uppercase hover:text-green duration-200",
      ...(href && { href }),
    },
    <>
      {children}
      <span className="absolute bottom-0 left-0 h-[2px] w-full bg-green" />
    </>,
  );
```

# My temp solution:
```js

  const className =
    "relative mx-auto inline-block h-[38px] font-bold leading-7 tracking-[2.29px] md:leading-[26px] uppercase hover:text-green duration-200";

  return React.createElement(
    TagName,
    {
      className,
      ...(href && { href }),
    },
    <>
      {children}
      <span className="absolute bottom-0 left-0 h-[2px] w-full bg-green" />
    </>,
  );
```

**Comment by wongjn** [comment]
Seems to work [in this Stackblitz reproduction of your code](https://stackblitz.com/edit/tailwindlabs-tailwindcss-13602?file=src%2Fmain.jsx). Perhaps it is something else in your project? Could you provide a project that reproduces the unexpected behavior please?

**Reply by 0x41-li** [reply]
Thank you for the response.

Strange.

Here's the project with the issue:

https://github.com/0x41-li/single-page-developer-portfolio

Note: it has the temp solution implemented, and the code is on the `components/buttons/Primary.tsx` file.

**Reply by 0x41-li** [reply]
While working on the project, I still have the same issue, I think this is an issue because I used the default config, and I didn't alter anything, now even the temp solution has requirements, the variable name should be named exactly `className`, any other name won't make the classes' styles get generated.

**Reply by wongjn** [reply, marked as answer]
Seems to work fine for me, after changing back to the object syntax you had before:
![image](https://github.com/tailwindlabs/tailwindcss/assets/11310624/c99f0ab4-7690-426b-8ae5-7fcd3ab80495)

Unfortunately, it seems like it could be something peculiar with your development environment that is stopping Tailwind from compiling the class rules, which is a lot harder to debug remotely 😅

---

Could it be that you are inquiring about the Tailwind Intellisense VSCode plugin? If so, you could try configuring the [`tailwindCSS.experimental.classRegex` setting](https://github.com/tailwindlabs/tailwindcss/issues/7553#issuecomment-735915659).

**Reply by 0x41-li** [reply]
Since it works on your side, it's definitely something on my dev env preventing Tailwind from compiling.  🫡

I want to thank you for your assistance, and your time.

I will check this more, and update this discussion if I find the root of the issue and solution.

**Comment by WaiiM** [comment]
Hi everyone !
I'm encountering the same issue when using the children prop. The Tailwind CSS classes like text-color, margin, and padding are not being generated. I've verified that Tailwind CSS is correctly installed and configured in my project. This problem seems to specifically occur when applying classes to elements rendered via the children prop.

**Reply by wongjn** [reply]
Consider checking that the file(s) that render the children are covered by the `content` file globs in the Tailwind configuration.
